### PR TITLE
Upgrade the runtime to the Substrate v2.0.0

### PR DESCRIPTION
**Pull request by shamil-gadelshin**
SSIA

### CODE REVIEW

File: runtime/src/lib.rs [hunk 1]
```diff
@@ -366,7 +413,7 @@ parameter_types! {
 }
 
 impl pallet_finality_tracker::Trait for Runtime {
-    type OnFinalizationStalled = Grandpa;
+    type OnFinalizationStalled = ();
```
Comment (mnaamani) [inline runtime/src/lib.rs:416]: What behavior do we loose or gain with this change?

Comment (shamil-gadelshin) [reply runtime/src/lib.rs:416]: Subsrate code for the finality tracker trait:
```
	/// Something which can be notified when the timestamp is set. Set this to `()`
	/// if not needed.
	type OnFinalizationStalled: OnFinalizationStalled<Self::BlockNumber>;
```
The Substrate template node runtime sets this to ().
Polkadot runtime doesn't contain the finality tracker at all.

File: runtime-modules/versioned-store/src/lib.rs [hunk 2]
```diff
@@ -10,6 +10,7 @@
 
 // Ensure we're `no_std` when compiling for Wasm.
 #![cfg_attr(not(feature = "std"), no_std)]
+#![allow(clippy::match_like_matches_macro)]
```
Comment (mnaamani) [inline runtime-modules/versioned-store/src/lib.rs:13]: I think I just need a newer version of clippy if I'm seeing this?

```
Checking pallet-versioned-store v3.1.0 (/joystream/runtime-modules/versioned-store)
error: unknown clippy lint: clippy::match_like_matches_macro
  --> runtime-modules/versioned-store/src/lib.rs:13:10
   |
13 | #![allow(clippy::match_like_matches_macro)]
   |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `-D clippy::unknown-clippy-lints` implied by `-D warnings`
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unknown_clippy_lints

error: aborting due to previous error
```

**Review by conectado — APPROVED** [review]

**Review by mnaamani — APPROVED** [review]
build and tested with `nightly-2020-10-06` for WASM build and rustc 1.47.0 for native build.

**Pull request merged**